pcsx2-qt/Settings/AchievementSettingsWidget.cpp

```diff
@@ -0,0 +1,195 @@
+	dialog->registerWidgetHelp(m_ui.challengeMode, tr("Enable Hardcore Mode"), tr("Unchecked"),
+		tr("\"Challenge\" mode for achievements. Disables save state, cheats, and slowdown "
+		   "functions, but you receive double the achievement points."));
+	dialog->registerWidgetHelp(m_ui.leaderboards, tr("Enable Leaderboards"), tr("Checked"),
+		tr("Enables tracking and submission of leaderboards in supported games. If leaderboards are disabled, you will still "
+		   "be able to view the leaderboard and scores, but no scores will be uploaded."));
+	dialog->registerWidgetHelp(
+		m_ui.soundEffects, tr("Enable Sound Effects"), tr("Checked"),
+		tr("Plays sound effects for events such as achievement unlocks and leaderboard submissions."));
+
+	connect(m_ui.enable, &QCheckBox::stateChanged, this, &AchievementSettingsWidget::updateEnableState);
+	connect(m_ui.challengeMode, &QCheckBox::stateChanged, this, &AchievementSettingsWidget::updateEnableState);
+	connect(m_ui.challengeMode, &QCheckBox::stateChanged, this, &AchievementSettingsWidget::onChallengeModeStateChanged);
+
+	if (!m_dialog->isPerGameSettings())
+	{
+		connect(m_ui.loginButton, &QPushButton::clicked, this, &AchievementSettingsWidget::onLoginLogoutPressed);
+		connect(m_ui.viewProfile, &QPushButton::clicked, this, &AchievementSettingsWidget::onViewProfilePressed);
+		connect(g_emu_thread, &EmuThread::onAchievementsRefreshed, this, &AchievementSettingsWidget::onAchievementsRefreshed);
+		updateLoginState();
+
+		// force a refresh of game info
+		Host::RunOnCPUThread(Host::OnAchievementsRefreshed);
+	}
+	else
+	{
+		// remove login and game info, not relevant for per-game
+		m_ui.verticalLayout->removeWidget(m_ui.gameInfoBox);
+		m_ui.gameInfoBox->deleteLater();
+		m_ui.gameInfoBox = nullptr;
+		m_ui.verticalLayout->removeWidget(m_ui.loginBox);
+		m_ui.loginBox->deleteLater();
+		m_ui.loginBox = nullptr;
+	}
+
+	updateEnableState();
+}
+
+AchievementSettingsWidget::~AchievementSettingsWidget() = default;
+
+void AchievementSettingsWidget::updateEnableState()
+{
+	const bool enabled = m_dialog->getEffectiveBoolValue("Achievements", "Enabled", false);
+	const bool challenge = m_dialog->getEffectiveBoolValue("Achievements", "ChallengeMode", false);
+	m_ui.testMode->setEnabled(enabled);
+	m_ui.unofficialTestMode->setEnabled(enabled);
+	m_ui.richPresence->setEnabled(enabled);
+	m_ui.challengeMode->setEnabled(enabled);
+	m_ui.leaderboards->setEnabled(enabled && challenge);
+	m_ui.soundEffects->setEnabled(enabled);
+}
+
+void AchievementSettingsWidget::onChallengeModeStateChanged()
+{
+	if (!QtHost::IsVMValid())
+		return;
+
+	const bool enabled = m_dialog->getEffectiveBoolValue("Achievements", "Enabled", false);
+	const bool challenge = m_dialog->getEffectiveBoolValue("Achievements", "ChallengeMode", false);
+	if (!enabled || !challenge)
+		return;
+
+	// don't bother prompting if the game doesn't have achievements
+	auto lock = Achievements::GetLock();
```

Comment:
Not used?

Reply:
False positive from codacy. Anything which queries achievements states needs to hold the lock.